### ISSUE TITLE
Investigate static typing libraries

### ISSUE THREAD

**Issue by richardxia** [ShelterTechSF/askdarcel-web#230]
We should check out some static typing libraries such as [Flow](https://flow.org/) or [TypeScript](https://www.typescriptlang.org/) to help us catch certain types of errors before they get pushed to prod.

I don't have any personal experience with any JS static typing libraries, but whatever we pick should support gradual typing so that we don't need to add static typing hints to the whole codebase at once, and it needs to be compatible with React.

**Comment by richardxia** [comment]
I just learned that React has an optional library called [`prop-types`](https://facebook.github.io/react/docs/typechecking-with-proptypes.html) that allows you to define the schema for component props. The ESLint Airbnb ruleset requires `prop-types` by default, so I tried it out on one component in #235. If we think this is sufficient, we might not need to go all out with a JS static typing library.
